### PR TITLE
Enable/disable/coordinate spin per path

### LINKED ISSUE #2720: Privacy of spin with multiple connections on same 5 tuple
Text says:

> In case multiple connections share the same five-tuple, that is, have the same source and destination IP address and UDP ports, endpoints should try to co-ordinate across all connections to ensure a clear signal to any on-path measurement points.

How would that work when 1/8th of the connections are not spinning? Wouldn't that either be sending a misleading signal, or violate the 1/8th rule?

### PR DESCRIPTION
**Pull request by martinthomson**
Closes #2720.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -4019,14 +4019,14 @@ to disable the spin bit either globally or on a per-connection basis. Even when
...
 selection process SHOULD be designed such that on average the spin bit is
-disabled for at least one eighth of connections. The selection process performed
-at the beginning of the connection SHOULD be applied for all paths used by the
-connection.
-
-In case multiple connections share the same five-tuple, that is, have the same
-source and destination IP address and UDP ports, endpoints should try to
-co-ordinate across all connections to ensure a clear signal to any on-path
-measurement points.
+disabled for at least one eighth of network paths. The selection process
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: What happens when the client thinks the path hasn't changed but the server has?  An observer could be on either side of the NAT.

Comment (mikkelfj) [reply draft-ietf-quic-transport.md]: It doesn't really matter if the client and server disagrees in general as long as enough connections agree on enabling spin. As to privacy that is for justifying being silent so if one endpoint decides not to be, that shouldn't really matter either.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -4019,14 +4019,14 @@ to disable the spin bit either globally or on a per-connection basis. Even when
...
+performed at the beginning of the connection SHOULD be applied for all paths
+used by the connection.
+
+In case multiple connections share the same network path, as determined by
+having the same source and destination IP address and UDP ports, endpoints
+should try to co-ordinate across all connections to ensure a clear signal to any
+on-path measurement points.
```
Comment (ekr) [inline draft-ietf-quic-transport.md]: This seems like a classic case of RFC 6919 language. Are any of the implementations which (1) do spin and (2) reuse host ports intending to do this?

Comment (martinthomson) [reply draft-ietf-quic-transport.md]: I opened #2763 to explore that properly.

File: draft-ietf-quic-transport.md (hunk 1 above)
Comment (martinthomson) [reply draft-ietf-quic-transport.md]: Ian's question is somewhat relevant.  If the server changes their mind, then the spin state (spin, not) might change.  But that's OK.

**Pull request merged**